canonical-websites/www.ubuntu.com
# Update to latest run script

Update the `run` script and `package.json` to use latest patterns from [generator-canonical-webteam](https://www.npmjs.com/package/generator-canonical-webteam) version 0.1.6.

This involves updating the standard set of NPM scripts in `package.json`:

- build: Use `node-sass` to build sass from the `static/sass` dir into `static/css`
- watch: Watch for changes to `static/sass/styles.scss`
- test: Use `sass-lint` to check Scss against standard rules
- clean: Remove local files, e.g. `node_modules`

To accommodate these scripts, we are also:

- Moving `node-sass` and `ubuntu-vanilla-theme` from `dependencies` to `devDependencies`
- Using the standard version of `node-sass` from generator-canonical-webteam@0.1.6
- Adding `sass-lint` to `devDependencies`
- Updating `yarn.lock` to match these new dependencies
- Moving all `*.scss` files form `static/css` to `static/sass`
- Adding `.sass-lint.yml` sass linting rules from vanilla-framework

QA
--

Check out each of the functions of the run script:

``` bash
./run clean                   # Tidy up any previous settings
./run                         # Run the server
./run build                   # Build sass
./run watch                   # Watch for changes to sass files
./run serve --watch           # Run server with background watcher
./run serve --detach          # Run the server in the background 
./run serve --detach --watch  # Run the server in the background and watch
./run test                    # Test the sass config
```
